Sources/OpenFeature/Provider/ProviderEvents.swift

```diff
@@ -9,4 +9,5 @@ public enum ProviderEvent: String, CaseIterable {
     case error = "PROVIDER_ERROR"
     case configurationChanged = "PROVIDER_CONFIGURATION_CHANGED"
     case stale = "PROVIDER_STALE"
+    case notReady = "PROVIDER_NOT_READY"
```

Comment:
what's the difference between not ready and stale?

Reply:
Now that we are sort-of compliant I can point to the docs: https://openfeature.dev/specification/types/#provider-status

Reply:
You may be interested in [this GitHub issue](https://github.com/open-feature/spec/issues/238) regarding updating how provider state is managed.

Reply:
@beeme1mr thank you for the relevant pointer. I love the attempt in the Issue to further refine the set of Events. It also seems that everybody agrees on removing "STATE" as a concept exposed to the application authors, which is perfectly in line with this PR.

The one aspect of the Issue that is in opposition with our Kotlin/Swift SDK implementations, is the intended source generating Events: our SDKs expect the Provider to generate Events, while the proposal mentions:
```
 This will make the provider lifecycle entirely the domain of the SDK; the provider would need only to implement the optional initialize and shutdown, and context change (client-only) methods, and the SDK should handle the rest.
```

Reply:
We assessed that this PR is still an improvement until the proposal @beeme1mr linked gets consolidated